### DISCUSSION TITLE
Settings Page for Shell Script

### DISCUSSION THREAD

**Discussion by tassilo-net** [AxisCommunications/acap-native-sdk-examples#220] [Q&A] [answered]
Hi,

I'm trying to write a simple shell script that should periodically check a URL and play a sound for a defined HTTP response (it's an axis audio device).

The shell script with the URL query works. I would just like to have a GUI to enter the parameters.

I tried the hint in this answer https://github.com/AxisCommunications/acap-native-sdk-examples/discussions/163#discussioncomment-7400395. Unfortunately, I don't have a link to the HTML page for the apps in the Axis device. I have a Settings submenu, but there are no fields there.

I would be very happy about advice.

![grafik](https://github.com/AxisCommunications/acap-native-sdk-examples/assets/59343225/c3da98fb-2e10-4687-8b9c-b8b65671cf58)

Shell Script Base: https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/shell-script-example

**Comment by tassilo-net** [comment]
Excuse me. my mistake. I took hidden:string from another code in the paramConfig. That was the error of the blank seeing screen.

And the button for the html page is only displayed when the app is running.

Sorry. I could have figured that out myself.

I have one more question. What is the best way to query the app parameters via shell? I currently do this by calling /usr/local/packages/urlcheck/conf/urlcheck.conf
is this a safe way? Also when it comes to firmware updates?

**Reply by Cacsjep** [reply]
Would it not better todo this not via Just a Shell Script an using instead axparamter API? Because you will I think do changes when Parameter is updated right?

**Reply by tassilo-net** [reply]
That would certainly be better. However, I don't have time to deal with it at the moment.

I just want to read out the settings when starting the script.

**Reply by pataxis** [reply, marked as answer]
> is this a safe way? Also when it comes to firmware updates?

There's no supported way to query parameters from a shell script today.
The supported way to query parameters from an ACAP application is to use the solution in [axparameter](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/axparameter) or the one from [vapix](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/vapix) example. Both are examples in C.
